src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantDataManager.java

```diff
@@ -0,0 +1,452 @@
+            }
+            else
+                value = vc.getAttributeAsDouble( annotationKey, Double.NaN );
+            if( Double.isInfinite(value) ) { value = Double.NaN; }
+            if( jitter && annotationKey.equalsIgnoreCase(GATKVCFConstants.HAPLOTYPE_SCORE_KEY) && MathUtils.compareDoubles(value, 0.0, PRECISION) == 0 ) { value += 0.01 * Utils.getRandomGenerator().nextGaussian(); }
+            if( jitter && (annotationKey.equalsIgnoreCase(GATKVCFConstants.FISHER_STRAND_KEY) || annotationKey.equalsIgnoreCase(GATKVCFConstants.AS_FILTER_STATUS_KEY)) && MathUtils.compareDoubles(value, 0.0, PRECISION) == 0 ) { value += 0.01 * Utils.getRandomGenerator().nextGaussian(); }
+            if( jitter && annotationKey.equalsIgnoreCase(GATKVCFConstants.INBREEDING_COEFFICIENT_KEY) && MathUtils.compareDoubles(value, 0.0, PRECISION) == 0 ) { value += 0.01 * Utils.getRandomGenerator().nextGaussian(); }
+            if( jitter && (annotationKey.equalsIgnoreCase(GATKVCFConstants.STRAND_ODDS_RATIO_KEY) || annotationKey.equalsIgnoreCase(GATKVCFConstants.AS_STRAND_ODDS_RATIO_KEY)) && MathUtils.compareDoubles(value, LOG_OF_TWO, PRECISION) == 0 ) { value += 0.01 * Utils.getRandomGenerator().nextGaussian(); }   //min SOR is 2.0, then we take ln
+            if( jitter && (annotationKey.equalsIgnoreCase(VCFConstants.RMS_MAPPING_QUALITY_KEY) || annotationKey.equalsIgnoreCase(GATKVCFConstants.AS_RMS_MAPPING_QUALITY_KEY))) {
+                if( vrac.MQ_CAP > 0) {
+                    value = logitTransform(value, -SAFETY_OFFSET, vrac.MQ_CAP + SAFETY_OFFSET);
+                    if (MathUtils.compareDoubles(value, logitTransform(vrac.MQ_CAP, -SAFETY_OFFSET, vrac.MQ_CAP + SAFETY_OFFSET), PRECISION) == 0 ) {
+                        value += vrac.MQ_JITTER * Utils.getRandomGenerator().nextGaussian();
+                    }
+                } else if( MathUtils.compareDoubles(value, vrac.MQ_CAP, PRECISION) == 0 ) {
+                    value += vrac.MQ_JITTER * Utils.getRandomGenerator().nextGaussian();
+                }
+            }
+        } catch( Exception e ) {
+            //TODO: what exception is this handling ? it seems overly broad
+            value = Double.NaN; // The VQSR works with missing data by marginalizing over the missing dimension when evaluating the Gaussian mixture model
+        }
+
+        return value;
+    }
+
+    public void parseTrainingSets(
+            final FeatureContext featureContext,
+            final SimpleInterval targetLoc,
+            final VariantContext evalVC,
+            final VariantDatum datum,
+            final boolean TRUST_ALL_POLYMORPHIC ) {
+        datum.isKnown = false;
+        datum.atTruthSite = false;
+        datum.atTrainingSite = false;
+        datum.atAntiTrainingSite = false;
+        datum.prior = 2.0;
+
+        for( final TrainingSet trainingSet : trainingSets ) {
+            List<VariantContext> vcs = featureContext.getValues(trainingSet.variantSource, targetLoc.getStart());
```

Comment:
Is this right? The previous implementation just passed in `targetLoc` in its entirety, instead of just the start position. Should you be calling the `getValues(final FeatureInput<T> featureDescriptor, final SimpleInterval queryInterval)` overload instead?